### PR TITLE
🏁 Capstone Project: Final Review

### PR DESCRIPTION
**Pull request by Ruthmy**
# Rails Capstone Project `Rails` `PostgreSQL` `Rspec`

### Hi there! @Ruthmy here. 👋

This is the PR for the [Ruby on Rails capstone project - Budget app](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) milestone project. The goal of this project is to build a complete Rails application that allows users to create budgets and add expenses to them. Usershave a list of categories and can add expenses to each category. The app also provides insights into spending by category.
The project is built using the [Ruby on Rails](https://rubyonrails.org/) framework and [PostgreSQL](https://www.postgresql.org/) database.

### Design
- Adhered to design guidelines, including colors, typography, and layout.
- Used a design template by Gregoire Vella.

### Interactions

- Implemented a splash screen.
- Created sign-up and log-in pages.
- Designed a home page for categories.
- Developed a transactions page.
- Added "Add a new category" and "Add a new transaction" pages.

### Testing Requirements

- Created unit and integration tests for core components.

---

## **Project Goals:**

- [x] Used Postgres for the database. 
- [x] Develop a database schema that mirrors the structure of the Entity Relationship Diagram (ERD) provided in the project requirements.
- [x] Integrated Devise for authentication.
- [x] Validated all user inputs for security.
- [x] Chose a view template engine.
- [x] Deployed the app online.
- [x] Follow the [GitFlow as a workflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) to create and unify the different features of the project.
- [x] Adhere to established [best practices for Ruby programming](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md), ensuring code quality, readability, and maintainability.
- [x] Documented the work in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Implement unit and integration tests to verify the correctness of the codebase and promote code reliability.

---

[5f70b41f-d173-4ccb-8a22-a96ed3d799d9.webm](https://github.com/Ruthmy/budget-app/assets/15095218/4f1ba94b-bef0-4daa-8adf-e0044d8414de)

---

### Video: [Budget App](https://www.loom.com/share/a57b45a0708545d9bac18071ead813e2?sid=0cf18999-bcc6-4aa7-968e-4cb41812c951)

---

### 🌟 Thank you for your kind review 🌟

---

### CODE REVIEW

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Ruthmy ,
Your project is in review

**Comment by Munsa1** [comment]
- [ ] Hey @Ruthmy I cant seem to move past the splash screen what could be the problem?

**Comment by Ruthmy** [comment]
Hi @Munsa1, thanks for your comment.
Can you tell me if you are from local or from the deploy in render? Did you try clearing cookies and refreshing? Maybe use a different browser? In the video I used the deploy link to do the demo and it worked without problem. In my local it also managed to go from the splash to the login, sign up, etc screens.
Do you want to meet at zoom to review?

**Comment by Munsa1** [comment]
From local I get this
image

The app starts okay but I cannot login in or sign up

**Comment by Ruthmy** [comment]
@Munsa1 From where do you try to run it? I'm checking it on my computer again, give me 2min.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 <a name="readme-top"></a>
 
 <div align="center">
-<img src="./app/assets/images/logo_white.png" width="150px">
+<img src="./app/assets/images/logo_white.png" width="170px">
   <h1><b>Personal BudgetApp  | Ruby on Rails Capstone Project</b></h1>
 </div>
```
Comment (Munsa1) [inline README.md:6]: - [ ] Due to insufficient files changed I will leave a comment here. KIndly fix the routing issue as the app starts okay but I cannot sign up a new user or login due to a routing error. Due to this I am unable to check the app functionality

image
p/assets/81932478/5575cb03-aaf4-4279-8913-73241721108f)

### Kindly ensure you use the given font family as per requirement. I can see you have used a different one, while it looks good, we need to follow the given design in terms of the font family as well.

**Comment by Ruthmy** [comment]
@Munsa1 Hi again. I updated the manifest. Can you please try again??

**Comment by Munsa1** [comment]
Okay let me check

**Comment by Ruthmy** [comment]
Hello again @Munsa1 ! I tested again at my place and this time I recorded a video to show you how it works. Please let me know if you were able to access it.
https://www.loom.com/share/3164f9d7977e402e913b15de5877c272?sid=64c3e7d2-1aa3-45f0-98e2-2f8f6f033d5e

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Ruthmy, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._
#### :video_camera:  Video Presentation :film_projector: 
- [ ] Great Video presentation, you were audible and sounded clea. Your subject matter was clear and you showed the project functionality.

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Munsa1** [comment]
- [ ] Hey @Ruthmy  the user is still unable to create an account

**Comment by Ruthmy** [comment]
### Hi there! 
I appreciate your feedback @Munsa1 and the comments you have left to test the project. Here I will answer your comments and leave enough information for you and for the next code review.

- The screenshot you send me says `GET "site.webmanifest"`, this is NOT a path, so trying to access it will not work. This is a JSON file with application parameters (like name, author, icon and description). So this error does not indicate a routing problem within the application, since that route does not exist. If there is a routing problem, the application does not work locally, [as I showed you in the video I recorded for you](https://www.loom.com/share/3164f9d7977e402e913b15de5877c272?sid=64c3e7d2-1aa3-45f0-98e2-2f8f6f033d5e), and it would also not work when deploying to render, which you can see in t[he video presentation](https://www.loom.com/share/a57b45a0708545d9bac18071ead813e2?sid=0cf18999-bcc6-4aa7-968e-4cb41812c951) and the [video I recorded for the Pull Request](https://github.com/Ruthmy/budget-app/pull/9#:~:text=5f70b41f%2Dd173%2D4ccb%2D8a22%2Da96ed3d799d9.webm) text.

Screenshot_2

- About the font style. I have already edited the CSS import. But take into account that this [font is paid by Adobe](https://fonts.adobe.com/fonts/proxima-nova#fonts-section). For that reason although there is the import of the font, at the end it will load some of the following options that are declared in the CSS.

Thank you very much for everything and let me know if you need anything else.

**Review by peterrobert — APPROVED** [review]
Hi @Ruthmy , 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Your application is working fine now. Great work with your UI. It is working perfectly ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ruthmy** [comment]
### ✨✨Thank you very much @peterrobert ✨✨

**Pull request merged**